src/main/groovy/org/pih/warehouse/core/HistoryItem.groovy

```diff
@@ -0,0 +1,15 @@
+package org.pih.warehouse.core
+
+class HistoryItem<T> {
+    String identifier
```

Comment:
The details from the "reference document" probably move into the ~HistoryItemLink~ ~HistoryItemDetails~ ReferenceDocument class (id, name, URL) discussed below. 

> [!NOTE]  
> A reference document is a generic term used in ERPs and WMSs to generically define the object that 

I'm also fine with those properties living in the HistoryItem object but at some point (when we do the same for stock history) we'll probably have two of these objects (one for the transaction, one for the reference document) so we'll want to refactor at that point anyway.

We probably don't need an identifier here, just the URL (which requires the UUID) and label (which requires the identifier) used for displaying the anchor tag. So here's how i would probably model the generic history item for now.

HistoryItem
```
Date date
Location location
ReferenceDocument referenceDocument
```
ReferenceDocument (alternatives: HistoryItemLink, HistoryItemDetails)
```
// Required to render a link
String label // label to display with URL
String url // URL for the link

// These might be useful, but probably aren't necessary
String id  // uuid assigned by database
String identifier  // human readable identifier
String name  // long human unreadable name
String description // extra description if necessary
```

The actual reference document class (Shipment, Receipt) should be responsible for generating these objects because they will know how to generate a link to the correct resource (i.e. stock movement details page)
```
<a href="${historyItem.referenceDocument.url}">${historyItem.referenceDocument.label}</a>
```

We're already doing this, but just wanted to show why it's necessary and suggest there might be room for improvement here. For example, the Shipment.getHistory() is pretty gnarly (my fault) so maybe there's a way to delegate some of that work to the dependent objects.